sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -251,7 +253,11 @@ def send_options_request(max_retries=5):
           # This reports channel is READY but connections may fail
           # Seems to be only an issue on Mac with port forwardings
           return job_service.DescribePipelineOptions(
-              beam_job_api_pb2.DescribePipelineOptionsRequest())
+              beam_job_api_pb2.DescribePipelineOptionsRequest(),
+                  timeout=portable_options.job_server_timeout)
```

Comment:
I think this line is causing tests to fail. Also, we're already retrying this request. So I would either keep the existing logic or add the new timeout, but not both.

Reply:
My guess is that this is invoking the LocalJobServicer that doesn't have the timeout argument. I'll try to repro locally to verify. If that's the case I'd like to add the timeout argument to all LocalJobServicer methods.

Reply:
I fixed the tests by adding the timeout to the stubs of the LocalJobServicer.

Reply:
Adding the timeout to the stubs fixes the tests, but are you sure adding the timeout like that is actually effective? In all the other places you changed, you put timeout in the requests, but here you put it in the service method, which I assumed was the problem. ie `DescribePipelineOptions(DescribePipelineOptionsRequest(), timeout=...)` should be  `DescribePipelineOptions(DescribePipelineOptionsRequest(timeout=...))`

Reply:
hmmm, I may be missing something but the timeout is never added to the request. My understanding is that timeout itself is not a request parameter but rather a grpc method call parameter. @ibzib can you point me to the code where you see me adding the timeout to the request? Perhaps, I accidentally added it to the wrong place. 
I'm no grpc expert, but I've ran some manual tests on this feature:
1) run python driver without job server running and default timeout -> The driver times out as expected after 60 seconds.
2) run python driver without job server running and change timeout -> The driver times out as expected after the specified timeout.
3) run python driver with job server running but code changed to slow down Prepare() -> The driver times out as expected.
4) run python driver with long running streaming job (Run() method) -> the driver program never times out (note this is the expected behavior since the driver is stuck waiting for completion.

Reply:
Oh, I read everything wrong. :sweat_smile: 

It'd be great to have real tests to verify all these timeouts, but I guess they would not make suitable unit tests, as they would all require at least 1 second to complete.